fake-server/index.js

```diff
@@ -0,0 +1,31 @@
+const express = require("express")
+const bodyParser = require('body-parser');
+const fakeUsers = require("./fake-users.json");
+const app = express();
+const port = 3000
+
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+
+app.post("/", (req, res) => {
+    const { username, password } = req.body || {};
+    var policy = "";
+    if(!username || !password) {
+        res.status(400).send("Bad request");
+    } else {

```

Comment:
I'm finding these nested if statements a little tricky to read. Maybe they could be refactored and split into functions? For each response.